psp/data_sources/nwp.py

```diff
@@ -63,7 +64,11 @@ def _slice_on_lat_lon(
 
     elif nearest_lat is not None and nearest_lon is not None:
         ((x, y),) = transformer([(nearest_lat, nearest_lon)])
-        return data.sel(x=x, y=y, method="nearest")  # type: ignore
+
+        if not single_point:
+            return data.sel(x=x, y=y, method="nearest")  # type: ignore
```

Comment:
This actually returns a single point!
In the `else` it just doesn't use `nearest_{lat,lon}`. This looks like a bug.